tests/group_functions/group_barrier.cpp

```diff
@@ -0,0 +1,234 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include <catch2/catch_template_test_macros.hpp>
+
+#include "group_functions_common.h"
+
+template <int D>
+class test_fence;
+
+TEMPLATE_TEST_CASE_SIG("Group barriers", "[group_func][dim]", ((int D), D), 1,
+                       2, 3) {
+  auto queue = sycl_cts::util::get_cts_object::queue();
+
+  // FIXME: hipSYCL and DPCPP have no implemented
+  //  atomic_fence_scope_capabilities query
+#if defined(SYCL_CTS_COMPILING_WITH_HIPSYCL) || \
+    defined(SYCL_CTS_COMPILING_WITH_DPCPP)
+  // mock for not available info
+  std::vector<sycl::memory_scope> supported_barriers{
+      sycl::memory_scope::sub_group, sycl::memory_scope::work_group,
+      sycl::memory_scope::device, sycl::memory_scope::system};
+  WARN(
+      "hipSYCL and DPCPP have no implementation of "
+      "atomic_fence_scope_capabilities query, suppose all barrier types as "
+      "valid.");
+#else
+  std::vector<sycl::memory_scope> supported_barriers =
+      queue.get_context()
+          .get_info<sycl::info::context::atomic_fence_scope_capabilities>();
+#endif
+
+  using sms = std::tuple<sycl::memory_scope, bool, bool>;
+  enum s { scope = 0, support = 1, test = 2 };
+
+  std::array<sms, 4> group_barriers{
+      sms{sycl::memory_scope::work_group, true, true},
+      sms{sycl::memory_scope::work_group, true, true},
+      sms{sycl::memory_scope::device, true, true},
+      sms{sycl::memory_scope::system, true, true}};
```

Comment:
Round 2, sorry for the noise
```suggestion
  std::array<sms, 4> group_barriers {{
      {sycl::memory_scope::work_group, true, true},
      {sycl::memory_scope::work_group, true, true},
      {sycl::memory_scope::device, true, true},
      {sycl::memory_scope::system, true, true}}};
```

Reply:
Done.